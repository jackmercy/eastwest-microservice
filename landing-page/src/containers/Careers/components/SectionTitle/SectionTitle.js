import Link from 'next/link';
import { memo, useCallback } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { useIntl } from 'react-intl';

import { scrollToElement } from '../../../../utils';

import useStyles from './sectionTitle.styles';

const SectionTitle = ({ id, translateId, className }) => {
  const intl = useIntl();
  const classes = useStyles();

  const handleScrollToId = useCallback((e) => {
    e.preventDefault();
    window.history.pushState({}, '', `/careers#${id}`);
    scrollToElement(id);
  }, []);

  return (
    <Link href={`/careers#${id}`}>
      <h2
        id={id}
        role="presentation"
        className={classNames(classes.title, className)}
        onClick={handleScrollToId}
      >
        {intl.formatMessage(
          { id: translateId },
          {
            span: (...chunks) => (
              <span className={classes.primary}>{chunks.join('')}</span>
            ),
          },
        )}
      </h2>
    </Link>
  );
};
SectionTitle.propTypes = {
  id: PropTypes.string.isRequired,
  translateId: PropTypes.string.isRequired,
  className: PropTypes.string,
};
SectionTitle.defaultProps = { className: '' };

export default memo(SectionTitle);
