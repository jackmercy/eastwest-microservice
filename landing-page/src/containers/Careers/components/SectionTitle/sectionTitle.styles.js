import { createUseStyles } from 'react-jss';

import { rem } from '../../../../utils/jss';
import COLORS from '../../../../constants/COLORS';
import BREAKPOINTS from '../../../../constants/BREAKPOINTS';

export default createUseStyles({
  primary: { color: COLORS.primary },
  title: {
    fontWeight: 900,
    fontSize: rem(56),
    lineHeight: rem(72),
    cursor: 'pointer',
    marginTop: 0,
    marginBottom: rem(24),
  },

  [`@media only screen and (max-width: ${BREAKPOINTS.maxSm}px)`]: {
    title: {
      textAlign: 'center',
      fontSize: rem(40),
      lineHeight: rem(48),
    },
  },
});
