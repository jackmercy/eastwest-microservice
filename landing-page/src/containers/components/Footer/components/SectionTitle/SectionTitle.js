import React, { memo } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import useStyles from './sectionTitle.styles';

const SectionTitle = ({ title, className, dataTest }) => {
  const classes = useStyles();

  return (
    <h2 className={classNames(classes.root, className)} data-test={dataTest}>
      {title}
    </h2>
  );
};
SectionTitle.propTypes = {
  title: PropTypes.node.isRequired,
  className: PropTypes.string,
  dataTest: PropTypes.string,
};
SectionTitle.defaultProps = { className: '', dataTest: '' };

export default memo(SectionTitle);
