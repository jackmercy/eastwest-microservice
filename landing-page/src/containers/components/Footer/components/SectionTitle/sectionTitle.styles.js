import { createUseStyles } from 'react-jss';

import { rem } from '../../../../../utils/jss';
import COLORS from '../../../../../constants/COLORS';

export default createUseStyles({
  root: {
    fontWeight: 900,
    fontSize: rem(18),
    lineHeight: rem(32),
    textTransform: 'uppercase',
    color: COLORS.textWhite,
    margin: 0,
  },
});
