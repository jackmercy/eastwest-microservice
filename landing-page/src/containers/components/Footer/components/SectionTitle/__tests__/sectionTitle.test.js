import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import SectionTitle from '../SectionTitle';

describe('Testing Footer > SectionTitle', () => {
  const props = { title: 'Section' };

  it('should render match snapshot', () => {
    const wrapper = shallow(<SectionTitle {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
