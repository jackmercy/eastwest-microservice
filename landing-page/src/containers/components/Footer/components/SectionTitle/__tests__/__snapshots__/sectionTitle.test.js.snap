// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Footer > SectionTitle should render match snapshot 1`] = `
<h2
  className="root-0-2-1"
  data-test=""
>
  Section
</h2>
`;
